src/main/java/com/ty/workSpace_Management/dao/WorkSpaceEntityDao.java

```diff
@@ -0,0 +1,60 @@
+package com.ty.workSpace_Management.dao;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import com.ty.workSpace_Management.entity.FloorEntity;
+import com.ty.workSpace_Management.entity.WorkSpaceEntity;
+import com.ty.workSpace_Management.repo.WorkSpaceRepo;
+
+@Repository
+public class WorkSpaceEntityDao {
+
+	@Autowired
+	private WorkSpaceRepo repo;
+
+	public WorkSpaceEntity saveWorkSpaceEntity(WorkSpaceEntity entity) {
+
+		return repo.save(entity);
+
+	}
+
+	public WorkSpaceEntity getWorkSpaceEntity(String workSpaceId) {
+		Optional<WorkSpaceEntity> floor = repo.findById(workSpaceId);
+		if (floor.isPresent()) {
+			return floor.get();
+		}
+		return null;
+	}
+
+	public WorkSpaceEntity updateWorkSpaceEntity(String workSpaceId, WorkSpaceEntity workSpaceEntity) {
+		WorkSpaceEntity entity = getWorkSpaceEntity(workSpaceId);
+		if (entity != null) {
+			workSpaceEntity.setWorkSpaceId(workSpaceId);
```

Comment:
try a code for checking capacity availability cost and type . check the object values if it  wont have the value try to set the old values